lib/pghero/methods/tables.rb

```diff
@@ -1,67 +1,294 @@
+            FROM
+              pg_statio_user_tables
+          SQL
+          )
+        else
+          select_one(<<-SQL
+            WITH worker_heap_blks_stats_json AS (
+              SELECT
+                result::json
+              FROM
+                run_command_on_workers($cmd$
+                SELECT
+                  json_agg(row_to_json(worker_heap_blks_stats))
+                FROM
+                  (
+                    SELECT
+                      sum(heap_blks_hit) AS heap_blks_hit,
+                      sum(heap_blks_read) AS heap_blks_read
+                    FROM
+                      pg_statio_user_tables
+                  )
+                  worker_heap_blks_stats $cmd$)
+            )
+            SELECT
+              sum(heap_blks_hit) / nullif(sum(heap_blks_hit) + sum(heap_blks_read), 0) AS rate
+            FROM
+              (
+                SELECT
+                  heap_blks_hit,
+                  heap_blks_read
+                FROM
+                  pg_statio_user_tables
+                UNION ALL
+                SELECT
+                  (json_array_elements(result)->>'heap_blks_hit')::bigint AS heap_blks_hit,
+                  (json_array_elements(result)->>'heap_blks_read')::bigint AS heap_blks_read
+                FROM
+                  worker_heap_blks_stats_json
+              ) AS cluster_heap_blks_stats
+          SQL
+          )
+        end
       end
 
       def table_caching
-        select_all <<-SQL
-          SELECT
-            schemaname AS schema,
-            relname AS table,
-            CASE WHEN heap_blks_hit + heap_blks_read = 0 THEN
-              0
-            ELSE
-              ROUND(1.0 * heap_blks_hit / (heap_blks_hit + heap_blks_read), 2)
-            END AS hit_rate
-          FROM
-            pg_statio_user_tables
-          ORDER BY
-            2 DESC, 1
-        SQL
+        if !citus_enabled?
+          select_all <<-SQL
+            SELECT
+              schemaname AS schema,
+              relname AS table,
+              CASE WHEN heap_blks_hit + heap_blks_read = 0 THEN
+                0
+              ELSE
+                ROUND(1.0 * heap_blks_hit / (heap_blks_hit + heap_blks_read), 2)
+              END AS hit_rate
+            FROM
+              pg_statio_user_tables
+            ORDER BY
+              2 DESC, 1
+          SQL
+        else
+          select_all <<-SQL
+            WITH dist_heap_blks_stats AS (
+              WITH placements_heap_blks_stats AS (
+                SELECT
+                  logicalrelid,
+                  (run_command_on_placements(logicalrelid, $$
+                    SELECT
+                      heap_blks_hit AS hit
+                    FROM
+                      pg_statio_user_tables
+                    WHERE
+                      relid = '%s'::regclass::oid $$ )).result::bigint AS heap_blks_hit,
+                  (run_command_on_placements(logicalrelid, $$
```

Comment:
`run_command_on_placements` is much more expensive than `run_command_on_workers`, i.e. I'm eager to use multiple `run_command_on_workers` to simplify to queries but less eager to do it for `run_command_on_placements`.

How much this query would become more complex if we just use one `run_command_on_placements`? If it is too much we can keep it this way, if not let's use single `run_command_on_placements`.

Reply:
I come up with this one, maybe it can be simplified by the use of CTEs.

```sql
              SELECT
                logicalrelid,
                (json_array_elements(result::json)->>'heap_blks_hit')::bigint AS heap_blks_hit,
                (json_array_elements(result::json)->>'heap_blks_read')::bigint AS heap_blks_read
              FROM
                (SELECT
                  logicalrelid,
                  (run_command_on_placements(logicalrelid, $$
                    SELECT
                      json_agg(row_to_json(worker_heap_blks_stats))
                    FROM
                      (
                        SELECT
                          heap_blks_hit,
                          heap_blks_read
                        FROM
                          pg_statio_user_tables
                        WHERE
                          relid = '%s'::regclass::oid
                      )
                      worker_heap_blks_stats $$)).result
                FROM
                  pg_dist_partition) AS worker_heap_blks_stats;
```